tests/test_onnxml_decision_tree_converter.py

```diff
@@ -68,15 +68,17 @@ def _test_regressor(self, X, model, rtol=1e-06, atol=1e-06, extra_config={}):
 
     # Utility function for testing classification models.
     def _test_classifier(self, X, model, rtol=1e-06, atol=1e-06, extra_config={}):
-        onnx_ml_pred, onnx_pred, output_names = self._test_decision_tree(X, model, extra_config)
-
-        np.testing.assert_allclose(onnx_ml_pred[1], onnx_pred[1], rtol=rtol, atol=atol)  # labels
-        np.testing.assert_allclose(
-            list(map(lambda x: x if isinstance(x, np.ndarray) else list(x.values()), onnx_ml_pred[0])),
-            onnx_pred[0],
-            rtol=rtol,
-            atol=atol,
-        )  # probs
+        for n in [1, len(X)]:
```

Comment:
I don't think that this test is right. Should we convert the model once with X and then do predictions over the different batch sizes? As it is now this test just convert and run the model over the same batch size, it does not actual test that the sym dimension works.

Reply:
you're right, I've refactored the test harness to follow that logic.